File: backend/src/main/java/com/clinic/service/MedicineService.java
package com.clinic.service;

import com.clinic.dto.MedicineDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface MedicineService {
    MedicineDTO createMedicine(MedicineDTO medicineDTO);
    MedicineDTO updateMedicine(Long id, MedicineDTO medicineDTO);
    MedicineDTO getMedicineById(Long id);
    void deleteMedicine(Long id);
    Page<MedicineDTO> getAllMedicines(Pageable pageable);
    Page<MedicineDTO> searchMedicines(String keyword, Pageable pageable);
}

File: frontend/src/services/medicineService.js
import api from './api';

export const getMedicines = (page = 0, size = 10) => {
    return api.get(`/medicines?page=${page}&size=${size}`);
};

export const searchMedicines = (keyword, page = 0, size = 10) => {
    return api.get(`/medicines/search?keyword=${keyword}&page=${page}&size=${size}`);
};

export const getMedicineById = (id) => {
    return api.get(`/medicines/${id}`);
};

export const createMedicine = (medicineData) => {
    return api.post('/medicines', medicineData);
};

export const updateMedicine = (id, medicineData) => {
    return api.put(`/medicines/${id}`, medicineData);
};

export const deleteMedicine = (id) => {
    return api.delete(`/medicines/${id}`);
};
